### PR TITLE
Fixes:

### PR DESCRIPTION
**Pull request by ctpopescu**
- make sure we treat correctly composed types of Optional / List etc
in conceivable configuration
 - properly propagate the annotations from the actual base class
 - smaller cosmetic tune-ups

### CODE REVIEW

File: dataschema/schema2scala.py [hunk 1]
```diff
@@ -313,18 +312,21 @@ def _get_column_annotations(self, column: Schema.Column,
...
+            templates.append('Set')
+        elif column.is_repeated():
+            if column.info.repeated_semantics == Schema_pb2.ColumnInfo.REPEATED_ARRAY:
+                templates.append('Array')
+            elif column.info.repeated_semantics == Schema_pb2.ColumnInfo.REPEATED_SET:
+                templates.append('Set')
+            else:
+                templates.append('Seq')
+        if column.is_optional():
+            templates.append('Option')
```
Comment (vingkan) [inline dataschema/schema2scala.py:328]: Does the order in which templates are appended matter? Does Option need to come first?

File: dataschema/entity.py [hunk 2]
```diff
@@ -128,6 +129,17 @@ def GetStructuredTypeName(field_cls: type):
     return None
 
 
+def GetStructuredSubType(field_cls: type, index: int = 0):
+    """For a structured type return the subtype that it contains.
+    Normally for a list index would be 0, for a dict 0 for key, 1 for value."""
+    field_cls = GetOriginalType(field_cls)
+    if not hasattr(field_cls, '__origin__'):
```
Comment (vingkan) [inline dataschema/entity.py:136]: Will this still work if we upgrade to Python 3.7? I read in [this StackOverflow answer](https://stackoverflow.com/a/52664522) that in Python 3.7 we have to check the class parameters/args.

**Review by vingkan — APPROVED** [review]
LGTM! Asked some questions for my own curiosity.

File: dataschema/schema2scala.py (hunk 1 above)
Comment (ctpopescu) [reply dataschema/schema2scala.py:328]: Actually it does e.g. py: Optional[List[int]] => Option[Seq[Long]] vs. List[Optional[int]] => Seq[Option[int]]

File: dataschema/entity.py (hunk 2 above)
Comment (ctpopescu) [reply dataschema/entity.py:136]: We are in python 3.7 or alter

**Pull request merged**